src/learn/perceptron.py

```diff
@@ -10,14 +10,25 @@
                     format='%(asctime)s %(levelname)s: %(message)s',
                     datefmt='%m/%d/%Y %I:%M:%S %p')
 
-class PerceptronLearner():
+class Learner():
 
     def __init__(self, w_vector=None, max_iter=1):
+        """
+        :param w_vector: A global weight vector instance that stores
+         the weight value (float)
+        :param max_iter: Maximum iterations for training the weight vector
+         Could be overridden by parameter max_iter in the method
+        :return: None
+        """
         logging.debug("Initialize PerceptronLearner ... ")
         self.w_vector = w_vector
         self.max_iter = max_iter
+
         return
 
+    def learner_name(self):
```

Comment:
Sorry I meant instance methods.
We don't normally let other functions access class attributes right?